src/test/java/uk/gov/ons/ctp/response/casesvc/endpoint/CaseActionEventIT.java

```diff
@@ -0,0 +1,247 @@
+    Assert.assertEquals(202, processEventResponse.getStatus());
+    Thread.sleep(5000);
+    List<CaseActionEventRequest> caseActionEventRequestList =
+        actionEventRequestRepository.findByCollectionExerciseIdAndEventTag(
+            collectionExerciseId, "mps");
+    Assert.assertEquals(1, caseActionEventRequestList.size());
+    Assert.assertEquals(
+        CaseActionEventRequest.ActionEventRequestStatus.COMPLETED,
+        caseActionEventRequestList.get(0).getStatus());
+    CaseActionEvent eventStatus = message.getPubSubCaseActionEventStatus();
+    Assert.assertEquals(
+        CaseActionEventRequest.ActionEventRequestStatus.COMPLETED, eventStatus.getStatus());
+  }
+
+  @Test
+  public void testProcessEventFailureScenario() throws Exception {
+    Mockito.when(
+            processEmailActionService.processEmailService(
+                any(CollectionExerciseDTO.class), anyString(), any(Instant.class)))
+        .thenReturn(new AsyncResult<>(false));
+    Mockito.when(
+            processLetterActionService.processLetterService(
+                any(CollectionExerciseDTO.class), anyString(), any(Instant.class)))
+        .thenReturn(new AsyncResult<>(true));
+    TestPubSubMessage message = new TestPubSubMessage();
+    UUID collectionExerciseId = createCollectionData();
+    UUID sampleUnitId = UUID.randomUUID();
+    caseCreator.postSampleUnit("LMS0001", "H", sampleUnitId, collectionExerciseId);
+    Thread.sleep(2000);
+    // When
+    CaseActionEvent caseActionEvent = new CaseActionEvent();
+    caseActionEvent.setCollectionExerciseID(collectionExerciseId);
+    caseActionEvent.setTag(CaseActionEvent.EventTag.go_live);
+    HttpResponse processEventResponse =
+        Unirest.post("http://localhost:" + port + "/process-event")
+            .basicAuth("admin", "secret")
+            .header("Content-Type", "application/json")
+            .body(caseActionEvent)
+            .asString();
+    Assert.assertEquals(202, processEventResponse.getStatus());
+    Thread.sleep(5000);
+    List<CaseActionEventRequest> caseActionEventRequestList =
+        actionEventRequestRepository.findByCollectionExerciseIdAndEventTag(
+            collectionExerciseId, "go_live");
+    Assert.assertEquals(1, caseActionEventRequestList.size());
+    Assert.assertEquals(
+        CaseActionEventRequest.ActionEventRequestStatus.RETRY,
+        caseActionEventRequestList.get(0).getStatus());
+    CaseActionEvent eventStatus = message.getPubSubCaseActionEventStatus();
+    Assert.assertEquals(
+        CaseActionEventRequest.ActionEventRequestStatus.RETRY, eventStatus.getStatus());
+    HttpResponse retryEventResponse =
+        Unirest.post("http://localhost:" + port + "/retry-event")
+            .basicAuth("admin", "secret")
+            .header("Content-Type", "application/json")
+            .body(caseActionEvent)
+            .asString();
+    Assert.assertEquals(202, retryEventResponse.getStatus());
+    Thread.sleep(5000);
```

Comment:
general question 5 seconds, 2 seconds above.
If they system is slow do these have an impact

Reply:
These are test cases. I believe the test makes use of single thread to carry on process. Hence I need to stop the processing to provide enough time for the docker image to process the request. Hence 5 sec.